client/src/components/modals_and_popovers/AriaModal.js

```diff
@@ -0,0 +1,57 @@
+import classNames from "classnames";
+import React from "react";
+import { Modal } from "react-bootstrap";
+import FocusLock from "react-focus-lock";
+
+import "./bootstrap_modal_exstension.scss";
+
+export class AriaModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    if (!props.titleId) {
+      throw new Error("Must have a prop titleId");
+    }
```

Comment:
Is using a separate, id'ed, component plus `aria-labelledby` just a holdover from the old `AriaModal`? Doesn't seem like it's the most flexible setup, probably more sense to give the modal an optional header (`Modal.Header` component with `react-bootstrap`) or just pass in a title string.

In fact, I don't see the RPB's supposed `#tbp-title` element in our source code so it may be unused. In that sense, this check's not actually very strong because it doesn't guarantee that a valid DOM node with id=`titleID` exists (or will exist, at render time). If this component wants to enforce a proper ARIA label then it would have to request it as a string and handle the labeling itself.

Reply:
yea the titleId and aria-labelledby was taken from the old AriaModal.

As for `#tbp-title`, I managed to find it being used in `Tablepicker.js`

Reply:
I think i'll go with the handling label by itself

Reply:
so it seems like either it will

a) focus on the modal itself, which doesn't have proper focus trapping but will read the aria label
b) focus immediately on the first tabbable child, which means aria-label is skipped

I tried using the `Modal.Header` but doesn't seem to help...

Reply:
~How about try and aim for b and make sure the `Modal.Header` is tabbable (if not directly then wrap it in a tabbable element)? I think the header's text should usually be the same as the `aria-label`, but if there's a case for needing additional descriptiveness for screen readers then `aria-label` could go on that first focusable (as long as it's not just a repeat of the header content).~

Oh, that's likely just a matter of making a new, markdown free, text key to use for the label. The alternate hacky approach is to strip the markup from the string. If you want to go with hacky, look up `DOMPurify`. We're already using it to sanitize input received from Titan that can contain HTML, but it can also just fully strip all HTML from a string, I believe.